src/main.rs

```diff
@@ -1,143 +1,68 @@
-                    response,
-                } => match response {
-                    CommandResponse::Pong => {
-                        println!("Received Pong for request {:?}", request_id);
-                    }
-                    CommandResponse::Other(result) => {
-                        println!(
-                            "Received Result for request {:?}: {:?}",
-                            request_id,
-                            String::from_utf8(result)
-                        );
-                    }
-                },
-            },
-            RequestResponseEvent::OutboundFailure {
-                peer,
-                request_id,
-                error,
-            } => println!(
-                "Outbound Failure for request {:?} to peer: {:?}: {:?}",
-                request_id, peer, error
-            ),
-            RequestResponseEvent::InboundFailure {
-                peer,
-                request_id,
-                error,
-            } => println!(
-                "Inbound Failure for request {:?} to peer: {:?}: {:?}",
-                request_id, peer, error
-            ),
-        }
-    }
-}
+type P2PNetworkSwarm = ExpandedSwarm<
+    P2PNetworkBehaviour,
+    <<<P2PNetworkBehaviour as NetworkBehaviour>::ProtocolsHandler as IntoProtocolsHandler>::Handler as ProtocolsHandler>::InEvent,
+    <<<P2PNetworkBehaviour as NetworkBehaviour>::ProtocolsHandler as IntoProtocolsHandler>::Handler as ProtocolsHandler>::OutEvent,
+    <P2PNetworkBehaviour as NetworkBehaviour>::ProtocolsHandler,
+    PeerId,
+>;
 
 fn main() -> Result<(), Box<dyn Error>> {
     // Create a random PeerId
-    let local_keys = identity::Keypair::generate_ed25519();
+    let local_keys = Keypair::generate_ed25519();
     let local_peer_id = PeerId::from(local_keys.public());
     println!("Local peer id: {:?}", local_peer_id);
 
     // create a transport
-    let transport = libp2p::build_development_transport(local_keys)?;
+    let transport = build_development_transport(local_keys)?;
 
-    // Create a Swarm that establishes connections through the given transport
-    // Use custom behaviour P2PNetworkBehaviour
-    let mut swarm = {
-        // Create a Kademlia behaviour.
+    // Create a Kademlia behaviour.
+    let kademlia = {
         let store = MemoryStore::new(local_peer_id.clone());
-        let kademlia = Kademlia::new(local_peer_id.clone(), store);
-        let mdns = Mdns::new()?;
+        Kademlia::new(local_peer_id.clone(), store)
+    };
+    let mdns = Mdns::new()?;
+
+    // Create RequestResponse behaviour with CommandProtocol
+    let msg_proto = {
         // set request_timeout and connection_keep_alive if necessary
         let cfg = RequestResponseConfig::default();
         let protocols = iter::once((CommandProtocol(), ProtocolSupport::Full));
-        let msg_proto = RequestResponse::new(CommandCodec(), protocols.clone(), cfg);
+        RequestResponse::new(CommandCodec(), protocols.clone(), cfg)
```

Comment:
extraneous clone here on protocols.  If you run `cargo clippy --all-targets --all-features -- -D warnings` you can catch many of these errors.